text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/nonstandard/CSSColorTagResolver.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.minimessage.tag.nonstandard;
+
+import java.util.HashMap;
+import java.util.Map;
+import net.kyori.adventure.text.format.TextColor;
+import net.kyori.adventure.text.minimessage.Context;
+import net.kyori.adventure.text.minimessage.ParsingException;
+import net.kyori.adventure.text.minimessage.tag.Tag;
+import net.kyori.adventure.text.minimessage.tag.resolver.ArgumentQueue;
+import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A transformation applying a single css text color.
+ *
+ * @since 4.10.0
+ */
+@ApiStatus.Internal
+public class CSSColorTagResolver implements TagResolver {
```

Comment:
I'd kinda like to expose API for users to build their own color tags with any set of aliases -- maybe `StandardTags.color[Like?](Map<String, TextColor> aliases)`, or `StandardTags.color[Like?](String qualifier, Map<String, TextColor> aliases)`? That would get rid of a separate tag implementation class for just the CSS colours.

The tag implementation classes shouldn't be part of the API either way, there should be a factory method for them if this class has to stick around.

Reply:
I like the idea of these methods and have them implemented. My one question is regarding the second method with a qualifier. Using the css colors as an example, if a user adds that resolver to the MiniMessage instance and uses the `<aqua>` tag, the aqua value from css instead of Minecraft will be used. 

Should the method with the qualifier prioritize Minecraft color codes if the qualifier isn't present? If it doesn't I'm not entirely sure why we'd need a qualifier for colors. 

It can be made to work properly if the user adds the standard color resolver after the css color resolver, but that seems unintuitive since the standard color resolver is already included.

Reply:
Hmmm nevermind on this. I think this way may be the best. Even if css colors are added after then Minecraft colors can still be accessed with `<c:name>` and if no qualifier is given it will default to last resolver added.